src/foam/util/AddressUtil.java

```diff
@@ -87,12 +88,16 @@ public static String normalizeRegion(X x, String country, String regionCode) {
       EQ(Region.COUNTRY_ID, country),
       OR(
         EQ(Region.ISO_CODE, regionCode),
-        STARTS_WITH_IC(Region.NAME, regionCode)
+        STARTS_WITH_IC(Region.NAME, regionCode),
+        IN(regionCode, Region.ALTERNATIVE_NAMES)
       )
     )).select(new AbstractSink() {
       public void put(Object o, Detachable d) {
         Region region = (Region) o;
-        if ( region.getIsoCode().equals(regionCode) || region.getName().equalsIgnoreCase(regionCode) ) {
+        if ( region.getIsoCode().equals(regionCode) 
```

Comment:
Curious why the predicate and the sink are required?

Reply:
I believe it's to account for any partial region name matches we get from the query since we use the STARTS_WITH_IC predicate. It looks like the sink just makes it so we return on the first region we get a perfect match on.

Reply:
sorry im with joel ... the predicate has already provided only records that would match your sink condition?
is there any practical chance of multiple records being returned here?  otherwise find would work no?

Reply:
not sure the index would work for STARTS_WITH_IC